### ISSUE TITLE
Logstash 2.x event clone operation isn't deep enough

### ISSUE THREAD

**Issue by jsvd** [elastic/logstash#5567]
```
/tmp/logstash-2.3.3 % cat  config
input { generator { count => 1 add_field => { "[a][b][c]" => "hey" } type => "original" } }
filter {
  clone { clones => ["copy"] }
  if [type] == "copy" {
    mutate { replace => { "[a][b][c]" => "you" } }
  }
}
output {
  stdout {
    codec => plain { format => "type: %{type} -> [a][b][c]: %{[a][b][c]}
" }
  }
}
```

```
/tmp/logstash-2.3.3 % bin/logstash -f config
Settings: Default pipeline workers: 4
Pipeline main started
type: original -> [a][b][c]: you
type: copy -> [a][b][c]: you
```

Seems to affect all of Logstash 2.x and 1.5.x
This does not affect master.

**Comment by colinsurprenant** [comment]
ah. the Ruby Event#clone method is actually not deep ... see https://github.com/elastic/logstash/blob/612c9b8a6d7e75780919caa337baf5085e3c3c50/logstash-core-event/lib/logstash/event.rb#L95-L100

**Comment by jsvd** [comment]
this has been fixed in 47bab49b84c64165951766906e89f3f2098d089a, which will likely be included in logstash 2.4.1
